hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -150,6 +152,9 @@ public static HoodieWriteConfig createHoodieConfig(String schemaStr, String base
         .withCompactionConfig(HoodieCompactionConfig.newBuilder()
             .withPayloadClass(parameters.get(DataSourceWriteOptions.PAYLOAD_CLASS_OPT_KEY().key()))
             .withInlineCompaction(inlineCompact).build())
+        .withClusteringConfig(HoodieClusteringConfig.newBuilder()
+            .withInlineClustering(!asyncClusteringEnabled)
```

Comment:
isn't this supposed to be INLINE_CLUSTERING_PROP(hoodie.clustering.inline)? what happens if there is some contradiction among this existing property and the new property(hoodie.datasource.clustering.async.enable)?